src/call.c

```diff
@@ -1119,6 +1119,9 @@ int call_connect(struct call *call, const struct pl *paddr)
 			err = pl_strdup(&call->peer_uri, &addr.auri);
 		}
 
+		if (pl_isset(&addr.dname))
+			pl_strdup(&call->peer_name, &addr.dname);
```

Comment:
This is problematic because the display-name and userinfo part of a SIP URI are semantically different. Especially because display-names are often quoted strings which are no allowed in the userinfo part. Also, they can be completely different strings.

Setting the `peer_name` here is probably not needed anyways.

Reply:
E.g. `/dialdir "2201" <2201@localhost> sendrecv` will result in BareSIP trying to call `<sip:"2201"@localhost>`.
And `/dialdir "Alice" <bob@localhost> sendrecv` will result in BareSIP trying to call `<sip:"Alice"@localhost>`.

Reply:
We need to set `call->peer_name`. See `event.c`:
```c
		peerdisplayname = call_peername(call);
		if (peerdisplayname){
				err |= odict_entry_add(od, "peerdisplayname",
						ODICT_STRING, peerdisplayname);
		}
```

This solutions is taken from incoming call:
```c
	if (pl_isset(&msg->from.dname)) {
		err = pl_strdup(&call->peer_name, &msg->from.dname);
		if (err)
			return err;
	}
```

Parsing of the display name out of the SIP URI is done by `sip_address_decode()`.

Reply:
You're right, this line is not the problem. Nonetheless, the behavior is very concerning:

> `/dialdir "2201" <2202@localhost> sendrecv` will result in BareSIP trying to call `<sip:"2201"@localhost>`

Reply:
Added a commit to support also the short form with display name:
```
/dialdir "2201" <2202@localhost> sendrecv
```

Reply:
Yes, this line is not problematic at all. I ran into a parsing issue that you have fixed - substituting the display name for the userpart was never intended. Looks good to me.